pkg/cli/alpha/internal/update.go

```diff
@@ -0,0 +1,353 @@
+	if err := projectConfigFile.LoadFrom(yaml.DefaultPath); err != nil { // TODO: assess if DefaultPath could be renamed to a more self-descriptive name
+		return fmt.Errorf("fail to run command: %w", err)
+	}
+
+	// Determine which Kubebuilder version to use for the update
+	cliVersion := projectConfigFile.Config().GetCliVersion()
+
+	// Allow override of the version from PROJECT file via command line flag
+	if opts.FromVersion != "" {
+		// Ensure version has 'v' prefix for consistency with GitHub releases
+		if !strings.HasPrefix(opts.FromVersion, "v") {
+			opts.FromVersion = "v" + opts.FromVersion
+		}
+		log.Infof("Overriding cliVersion field %s from PROJECT file with --from-version %s", cliVersion, opts.FromVersion)
+		cliVersion = opts.FromVersion
+	} else {
+		log.Infof("Using CLI version from PROJECT file: %s", cliVersion)
+	}
+
+	// Download the specific Kubebuilder binary version for generating clean scaffolding
+	tempDir, err := opts.downloadKubebuilderBinary(cliVersion)
+	if err != nil {
+		return fmt.Errorf("failed to download Kubebuilder %s binary: %w", cliVersion, err)
+	}
+	log.Infof("Downloaded binary kept at %s for debugging purposes", tempDir)
+
+	// Create ancestor branch with clean state for three-way merge
+	if err := opts.checkoutAncestorBranch(); err != nil {
+		return fmt.Errorf("failed to checkout the ancestor branch: %w", err)
+	}
+
+	// Remove all existing files to create a clean slate for re-scaffolding
+	if err := opts.cleanUpAncestorBranch(); err != nil {
+		return fmt.Errorf("failed to clean up the ancestor branch: %w", err)
+	}
+
+	// Generate clean scaffolding using the old Kubebuilder version
+	if err := opts.runAlphaGenerate(tempDir, cliVersion); err != nil {
+		return fmt.Errorf("failed to run alpha generate on ancestor branch: %w", err)
+	}
+
+	// Create current branch representing user's existing project state
+	if err := opts.checkoutCurrentOffAncestor(); err != nil {
+		return fmt.Errorf("failed to checkout current off ancestor: %w", err)
+	}
+
+	// Create upgrade branch with new Kubebuilder version scaffolding
+	if err := opts.checkoutUpgradeOffAncestor(); err != nil {
+		return fmt.Errorf("failed to checkout upgrade off ancestor: %w", err)
+	}
+
+	// Create merge branch to attempt automatic merging of changes
+	if err := opts.checkoutMergeOffCurrent(); err != nil {
+		return fmt.Errorf("failed to checkout merge branch off current: %w", err)
+	}
+
+	// Attempt to merge upgrade changes into the user's current state
+	if err := opts.mergeUpgradeIntoMerge(); err != nil {
+		return fmt.Errorf("failed to merge upgrade into merge branch: %w", err)
+	}
+
+	return nil
+}
+
+// downloadKubebuilderBinary downloads the specified version of Kubebuilder binary
+// from GitHub releases and saves it to a temporary directory with executable permissions.
+// Returns the temporary directory path containing the binary.
+func (opts *Update) downloadKubebuilderBinary(version string) (string, error) {
+	cliVersion := version
```

Comment:
could not it be part of Update struct ?
What values we use always?
Would make sense we have a struct for them?